dandiapi/api/services/metadata/__init__.py

```diff
@@ -140,43 +160,23 @@ def _build_validatable_version_metadata(version: Version) -> dict:
     with transaction.atomic():
         # validating version metadata needs to lock the version to avoid racing with
         # other modifications e.g. aggregate_assets_summary.
-        version = (
-            Version.objects.filter(id=version.id, status=Version.Status.PENDING)
-            .select_for_update()
-            .first()
-        )
+        version_qs = Version.objects.filter(id=version_id).select_for_update()
+        current_version = version_qs.first()
 
         # It's possible for this version to get deleted during execution of this function.
         # If that happens *before* the select_for_update query, return early.
-        if version is None:
+        if current_version is None:
             logger.info('Version %s no longer exists, skipping validation', version_id)
             return
 
-        version.status = Version.Status.VALIDATING
-        version.save()
-
-        try:
-            validate(
-                _build_validatable_version_metadata(version),
-                schema_key='PublishedDandiset',
-                json_validation=True,
+        # If the version has since been modified, return early
+        if current_version.status != Version.Status.PENDING:
+            logger.info(
+                'Skipping validation for version %s due to concurrent modification', version_id
             )
-        except dandischema.exceptions.ValidationError as e:
-            logger.info('Error while validating version %s', version.id)
-            version.status = Version.Status.INVALID
-
-            validation_errors = _collect_validation_errors(e)
-            version.validation_errors = validation_errors
-            version.save()
-            return
-        except ValueError as e:
-            # A bare ValueError is thrown when dandischema generates its own exceptions, like a
-            # mismatched schemaVersion.
-            version.status = Version.Status.INVALID
-            version.validation_errors = [{'field': '', 'message': str(e)}]
-            version.save()
             return
-        logger.info('Successfully validated version %s', version.id)
-        version.status = Version.Status.VALID
-        version.validation_errors = []
-        version.save()
+
+        # Set to validating and continue
+        version_qs.update(status=Version.Status.VALIDATING)
```

Comment:
This won't ever get committed to the DB, right? Since this code is wrapped in `transaction.atomic`, and the status gets updated again immediately after this.

Reply:
That's true, although that problem already existed in the prior implementation. I'm not sure it makes sense to try to solve that here and now, since it might change the implementation significantly. As far as I can tell the `VALIDATING` status doesn't really get used much, if at all.

Reply:
FWIW, a similar thing (for asset `VALIDATING` status) is being tracked in #1653. Perhaps we should have a parallel issue for version `VALIDATING` as well.

Reply:
Sounds good. Considering it was an issue prior to this PR as well, I think this is fine as-is.